store/modules/playback/getters.js

```diff
@@ -22,4 +22,7 @@ export default {
   tooltips_delay(state) {
     return state.tooltips_delay;
   },
+  is_valid_barcode(state) {
```

Comment:
it is not necessary to create `getters` that just directly return a state. Best Vuex practice is to read from the state directly using `mapState` or some other choice of syntax.  Getters should only be used for cases where some sort of computation is applied to a state before returning it

Reply:
For readability purposes, I added the `is_valid_barcode(state) {`,  as we use in two components `PlateBarcode.vue` and `DesktopPlayersControls.vue`, additionally the value of Vuex variable `is_valid_barcode` is updated under the mutation of `set_barcode_number` in the playback store.

Reply:
the `set_barcode_number` mutation does not use this getter.  You can still access the state directly and have the code read the same in the rest of the `PlateBarcode` and `DesktopPlayerControls` components by just using `mapState` instead of `mapGetters`.  That is why the `mapState` function was created by the Vuex team.  If you like, you can even rename the local computed variable differently than the name of the state in the store when you use `mapState`.  The Vuex team provided all this functionality to ensure that there is no reason to have `getters` that purely return a state from the store and don't perform any additional computations.

Reply:
Sorry, I don't agree, if we remove the the function the the getter function the file 'playback.spec.js', testcases are failing you can comment out the function 'playback/getters.js' and verify.

Kindly see the log I get when I run the following unit test of playback.spec.js

`● store/playback › mutations › barcode validation and is_valid_barcode verification › Given a barcode scanned results in value ME20044099, When validation rule FAILS  or PASSES due All criteria matches, Then validation results set is_valid_barcode to true

    expect(received).toBe(expected) // Object.is equality

    Expected: true
    Received: undefined

      221 |         async (platecode, reason, valid) => {
      222 |           store.commit("playback/set_barcode_number", platecode);
    > 223 |           expect(store.getters["playback/is_valid_barcode"]).toBe(valid);
          |                                                              ^
      224 |         }
      225 |       );
      226 |     });

      at tests/unit/store/playback.spec.js:223:62

----------------------------------|---------|----------|---------|---------|------------------------
File                              | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s      
----------------------------------|---------|----------|---------|---------|------------------------
All files                         |    50.2 |    34.24 |   45.28 |    49.9 |                        
 js_utils                         |   26.64 |    17.65 |   27.27 |   26.67 |                        
  axios_helpers.js                |     100 |      100 |     100 |     100 |                        
  text_validation.js              |   24.39 |    19.35 |   27.27 |   24.22 | 26,43-57,74,83,132-401 
  waveform_data_formatter.js      |   23.61 |        0 |      20 |   23.94 | 32-249                 
 store                            |   55.56 |      100 |      20 |   55.56 |                        
  ApiService.js                   |   42.86 |      100 |       0 |   42.86 | 11-28                  
  index.js                        |     100 |      100 |     100 |     100 |                        
 store/modules/flask              |   85.19 |    68.18 |      70 |   84.91 |                        
  actions.js                      |      80 |    61.11 |   66.67 |      80 | 28-32,42,78            
  enums.js                        |     100 |      100 |     100 |     100 |                        
  getters.js                      |       0 |      100 |       0 |       0 | 5-8                    
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |     100 |      100 |     100 |     100 |                        
  url_regex.js                    |     100 |      100 |     100 |     100 |                        
 store/modules/playback           |   95.83 |      100 |   86.21 |   95.79 |                        
  actions.js                      |   98.55 |      100 |   91.67 |   98.55 | 40                     
  enums.js                        |     100 |      100 |     100 |     100 |                        
  getters.js                      |      50 |      100 |      50 |      50 | 17-23                  
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |     100 |      100 |     100 |     100 |                        
 store/modules/settings           |   16.67 |      100 |    6.25 |   11.76 |                        
  actions.js                      |       0 |        0 |       0 |       0 |                        
  getters.js                      |       0 |      100 |       0 |       0 | 3-32                   
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |       0 |      100 |       0 |       0 | 3-15                   
 store/modules/twentyfourcontrols |      60 |      100 |   33.33 |      50 |                        
  getters.js                      |       0 |      100 |       0 |       0 | 3                      
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |       0 |      100 |       0 |       0 | 3                      
 store/modules/waveform           |      50 |    66.67 |   33.33 |   49.15 |                        
  actions.js                      |   42.31 |       50 |      50 |      44 | 22-54                  
  getters.js                      |   26.67 |      100 |       0 |   28.57 | 18-48                  
  index.js                        |     100 |      100 |     100 |     100 |                        
  mutations.js                    |   57.14 |      100 |   44.44 |   53.85 | 16-21,34-43            
  url_regex.js                    |     100 |      100 |     100 |     100 |                        
----------------------------------|---------|----------|---------|---------|------------------------
Test Suites: 1 failed, 1 total
Tests:       26 failed, 39 passed, 65 total
Snapshots:   0 total
Time:        3.023 s
Ran all test suites matching /tests\/unit\/store\/playback.spec.js/i.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @curi-bio/mantarray-frontend-components@0.1.2 test:unit: `npm run build && jest -c jest.config.js "tests/unit/store/playback.spec.js"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @curi-bio/mantarray-frontend-components@0.1.2 test:unit script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/ubuntu/.npm/_logs/2020-12-09T12_57_32_015Z-debug.log
raghu:~/environment/mantarray-frontend-components (plate_barcode_scanner_enhancements) $ `